### DISCUSSION TITLE
Is it possible to analyze replay file(.rep) data using this bwapi?

### DISCUSSION THREAD

**Discussion by capymind** [bwapi/bwapi#908] [Q&A] [answered]
Hi, this is awesome project!!

I am trying to create website for analyzing starcraft replay.

My main language is Python. (however I am damn beginner...)
In python community there is replay analyzer(file reader) only for Starcraft2

I want to implement something as follows (I don't know this is possible..)
- number of SCV during games
- average unit attack coherence?(hydrarisk moves all together or scatter)..
- something like that..

Can I learn the knowledge from this open source's to read starcraft replay files
and calculate basic summary statistics?  is there code section related to that?.

Thanks for reading a dumb.. question!

**Comment by heinermann** [comment, marked as answer]
There is an archived python project here: https://github.com/HearthSim/pyreplib/tree/master

I noticed some projects in C#, C++, and Go as well.

**Reply by capymind** [reply]
Oh... thanks for your kind comment. I will check it!!

**Comment by N00byEdge** [comment]
If I recall correctly, you can also write an AI module, attach it while watching replays to get all the game data during the game. You can also use the BWAPI AutoMenu to launch all your replays, one at a time

**Reply by capymind** [reply]
Wow.. it's cool. I will check it! thanks for sharing your knowledge. I hope I can make it.. anyway thanks again!
